cql3/query_processor.cc

```diff
@@ -771,27 +771,37 @@ query_options query_processor::make_internal_options(
                 format("Invalid number of values. Expecting {:d} but got {:d}", p->bound_names.size(), values.size()));
     }
     auto ni = p->bound_names.begin();
-    std::vector<cql3::raw_value> bound_values;
-    for (auto& v : values) {
-        auto& n = *ni++;
-        if (v.type() == bytes_type) {
-            bound_values.push_back(cql3::raw_value::make_value(value_cast<bytes>(v)));
-        } else if (v.is_null()) {
-            bound_values.push_back(cql3::raw_value::make_null());
-        } else {
-            bound_values.push_back(cql3::raw_value::make_value(n->type->decompose(v)));
-        }
+    raw_value_vector_with_unset bound_values;
+    bound_values.values.reserve(values.size());
+    bound_values.unset.resize(values.size());
+    for (auto& var : values) {
+        auto& n = *ni;
+        std::visit(overloaded_functor {
+            [&] (const data_value& v) {
+                if (v.type() == bytes_type) {
+                    bound_values.values.emplace_back(cql3::raw_value::make_value(value_cast<bytes>(v)));
```

Comment:
Note: this special case is probably a pessimization, since it forces linearization. It's not critical since I don't expect huge values here, but it would be nice to remove it, separately.